packages/govern-console/src/containers/CreateDao/CreateDaoProgress.tsx

```diff
@@ -42,22 +42,27 @@ const CreateDaoProgress: React.FC<{
   const [showAction, setShowAction] = useState<'none' | 'fail' | 'register' | 'finish'>('none');
   const [rule, SetRule] = useState<BytesLike>('');
   const [isNewDaoTokenRegistered, setIsNewDaoTokenRegistered] = useState(false);
-  const [daoTokenAddress, setDaoContractAddress] = useState('0x');
+  const [daoTokenAddress, setDaoTokenAddress] = useState('0x');
 
   const updateNewDaoTokenAddress = (value: string) => {
-    setDaoContractAddress(value);
+    setDaoTokenAddress(value);
   };
 
   useEffect(() => {
     const checkIfRegistered = async () => {
-      if (daoTokenAddress !== '0x') {
+      if (daoTokenAddress !== '0x' && progressList[1].status !== CiruclarProgressStatus.Done) {
         const isRegistered = await isTokenRegistered(provider.getSigner(), daoTokenAddress);
-        console.log('useEffect checkIfRegistered', daoTokenAddress, isRegistered);
         setIsNewDaoTokenRegistered(isRegistered);
+
+        // update create dao status
+        const newList2 = [...progressList];
+        newList2[1].status = CiruclarProgressStatus.Done;
+        setProgressList(newList2);
+        setShowAction('register');
       }
     };
     checkIfRegistered();
-  }, [daoTokenAddress, provider]);
+  }, [daoTokenAddress, provider, progressList]);
 
```

Comment:
not sure why `checkIfRegistered()` is checked as useEffect() instead of whenever daoTokenAddress is changed...

Reply:
i have that in mind for the next refactoring